app/main.py

```diff
@@ -1,6 +1,27 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        is_cachable = True
```

Comment:
I see you went an extra mile and added this check. Good job on that!

But on the other hand it is not required in task definition and overcomplicates the code, this function can be written in 6 lines with only difference that it will raise an exception if you try to add unhashable type to the dict as a key. 
As well when you perform your check it may slow down your code even more in case if you have a long list of arguments.